[glennliew] iP

glennliew opened the pull request:
# 🚀 Project Increment: Feature Update  

## Summary  
This PR introduces improvements to the ~~Duke~~ **Grennite** chatbot, including refactoring, bug fixes, and new features.  

## Changes Made  
- Refactored code using **OOP principles**  
- Implemented `find` command 🔍  
- Improved exception handling  


## Steps to Test  
1. Checkout this branch:  
   ```sh
   git checkout branch-Level-9

2. Run the program:
```sh java -jar Duke.jar```

3. Try adding tasks and using the `find` command

ToDo:
- [ ] Eat :dumpling:
- [ ] Sleep :sleeping:
- [x] Code :magic_wand:
- [x] Repeat :leftwards_arrow_with_hook:

> [!IMPORTANT]
> MUST SLEEP!


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

> Come take a look! [My Website](https://glennsportfolio.vercel.app/).

glennliew committed:
Level-0: Rename main class and add greeting

glennliew committed:
Level-1: Add support for echo functionality

glennliew committed:
Level-2: Add ability to store and list tasks

glennliew committed:
Level-3: Add ability to mark and unmark tasks

glennliew committed:
Add support for ToDos, Events, and Deadlines (Level-4)

glennliew committed:
Fix text UI testing format and resolve line-ending issues

glennliew committed:
Add DukeException class and enhance error handling

glennliew committed:
Level-6: Added delete functionality

glennliew committed:
A-Enums: Introduced Enums for task types

glennliew committed:
Implement Level 8: Dates and Times

glennliew committed:
Merge Level 8 feature

glennliew committed:
Refactored code using OOP (A-MoreOOP)

glennliew committed:
Refactored code using OOP (A-MoreOOP)

glennliew committed:
Merge Level 8 feature

glennliew committed:
Organize project structure into packages

glennliew committed:
Added Gradle support for project automation

glennliew committed:
Added JUnit tests for TaskList and Initialiazation

glennliew committed:
Packaged Grennite as an executable JAR

glennliew committed:
Add JavaDoc comments

glennliew committed:
Ensure code complies with coding standard

glennliew committed:
Implement Level-9 functionality

glennliew committed:
Merge tag 'branch-Level-9'

Implement search functionality

tadacheng commented on src/main/java/grennite/parser/Parser.java:
```diff
@@ -0,0 +1,82 @@
+import grennite.tasklist.TaskList;
+import grennite.ui.UI;
+
+public class Parser {
```
Perhaps you can use header comments to describe the class? same for other classes

tadacheng commented on src/main/java/grennite/parser/Parser.java:
```diff
@@ -0,0 +1,82 @@
+     * @return the index of the task
+     * @throws GrenniteException if the input is invalid
+     */
+    private int parseIndex(String[] words) throws GrenniteException {
```
Perhaps a more descriptive name like extractTaskIndex?

Codekrodile replied:
parseIndex() is a bit ambiguous, consider renaming it something like extractTaskIndex to better reflect its function

tadacheng commented on src/main/java/grennite/task/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+        this.deadline = LocalDateTime.parse(deadline, INPUT_FORMAT);
+    }
+
+    public Deadline(String description, String deadline, boolean isDone) {
```
I see that this constructor is repeated in deadline, event and todo class, is it possible to extract it to Task class so you do not need to repeat it

tadacheng commented on src/main/java/grennite/tasklist/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+import grennite.task.Todo; 
+import grennite.storage.Storage;
+
+public class TaskList {
```
I like your consistent method naming conventions

glennliew committed:
Apply CheckStyle checking

glennliew committed:
Add GUI

Codekrodile commented on src/main/java/grennite/parser/Parser.java:
```diff
@@ -0,0 +1,82 @@
+     */
+    public void processCommand(String input) throws GrenniteException {
+        String[] words = input.split(" ", 2);
+        String command = words[0];
```
consider checking words.length before access words

Codekrodile commented on src/main/java/grennite/task/Event.java:
```diff
@@ -0,0 +1,67 @@
+
+public class Event extends Task{
+    private LocalDate date;
+    private LocalTime from;
```
choice of variable name might be a bit confusing, consider changing it to fromDate and toDate

Codekrodile commented on src/main/java/grennite/tasklist/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+package grennite.tasklist;
+
+import java.time.LocalDate;
```
nit: should imports have lexicographically increasing order?

glennliew committed:
fix bugs

glennliew committed:
A-Assertions: Use assertions for validation

glennliew committed:
Use assertions for validation

Use Java's assert feature to document important assumptions in
Grennite and Parser classes.

- Added assertions to check non-null and non-empty filepath
  in Grennite constructor.
- Added assertions to verify non-null and non-blank input
  in Parser's processCommand method.

This helps catch logical errors during development.

glennliew committed:
Improve code quality

glennliew committed:
Fix search function bug

glennliew committed:
Add Duplicates function

glennliew committed:
Add UG and Ui.png

glennliew committed:
Merge pull request #2 from glennliew/Ui.png

Add UG and Ui.png

glennliew committed:
Update README.md

glennliew committed:
Update README.md

glennliew committed:
Update README.md

glennliew committed:
Fix README

glennliew committed:
Merge pull request #3 from glennliew/Ui.png

Fix README

glennliew committed:
"Fix initial message bug"

glennliew committed:
Update README.md

glennliew committed:
Update README.md
